edi_pdf2data/models/pdf2data_template.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2021 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import logging
+import os
+from tempfile import NamedTemporaryFile
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+try:
+    from invoice2data.extract.invoice_template import InvoiceTemplate
+    from invoice2data.extract.loader import ordered_load
+except ImportError:
+    _logger.debug("Cannot import invoice2data")
+
+try:
+    import fitz
+except ImportError:
+    _logger.debug("Cannot import PyMuPDF")
+
+
+class Pdf2dataTemplate(models.Model):
+    _name = "pdf2data.template"
+    _description = "Pdf2data Template"
+    _order = "sequence"
+
+    name = fields.Char()
+    code = fields.Char(related="type_id.code")
+    type_id = fields.Many2one("pdf2data.template.type", required=True)
+    pdf2data_template_yml = fields.Text()
+    pdf_file = fields.Binary(attachment=True)
+    pdf_filename = fields.Char()
+    file_result = fields.Char(readonly=True)
+    file_processed_result = fields.Html(readonly=True)
+    sequence = fields.Integer(default=20)
+    extracted_text = fields.Text(readonly=True)
+
+    def _parse_pdf(self, data):
+        data_file = NamedTemporaryFile(
```

Comment:
Wouldn't it make sense to move this to the specific component of each template?

Reply:
Not really, as the parsing depends directly on the template you are using, so, IMO, it seems legit to me.